### PR TITLE
Make IWebSocketResource methods noexcept

### PR DESCRIPTION
**Pull request by JunielKatarn**
`IWebSocketResource` implementations are already expected to signal error conditions through the provided error handler.
We should assure no exceptions will be thrown by this interface.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native-windows/pull/5487)

### CODE REVIEW

File: vnext/Shared/WinRTWebSocketResource.h [hunk 1]
```diff
@@ -47,89 +47,91 @@ class WinRTWebSocketResource : public IWebSocketResource, public std::enable_sha
...
       winrt::Windows::Foundation::Uri &&uri,
-      std::vector<winrt::Windows::Security::Cryptography::Certificates::ChainValidationResult> &&certExeptions);
+      std::vector<winrt::Windows::Security::Cryptography::Certificates::ChainValidationResult>
+          &&certExeptions) noexcept;
 
   WinRTWebSocketResource(
       const std::string &urlString,
-      std::vector<winrt::Windows::Security::Cryptography::Certificates::ChainValidationResult> &&certExeptions);
+      std::vector<winrt::Windows::Security::Cryptography::Certificates::ChainValidationResult>
+          &&certExeptions) noexcept;
```
Comment (NikoAri) [inline vnext/Shared/WinRTWebSocketResource.h:73]: >&&cert [](start = 10, length = 6)

(nit) it's a bit weird that this line is spit here

**Comment by NikoAri** [comment]
>   void EnqueueWrite(const std::string &message, bool binary);

Should these be noexcept too?

---
Refers to: vnext/Desktop/BeastWebSocketResource.h:58 in e0a2218. [](commit_id = e0a221837ae4aceab771d9f212da9ec61239c033, deletion_comment = False)

File: vnext/Shared/IWebSocketResource.h [hunk 2]
```diff
@@ -86,7 +86,7 @@ struct IWebSocketResource {
   static std::shared_ptr<IWebSocketResource>
   Make(const std::string &url, bool legacyImplementation = false, bool acceptSelfSigned = false);
```
Comment (NikoAri) [inline vnext/Shared/IWebSocketResource.h:87]: >Make [](start = 2, length = 4)

What about this one as noexcept?

Comment (NikoAri) [inline vnext/Shared/IWebSocketResource.h:87]: (nit) can we reformat this method to:
static std::shared_ptr<IWebSocketResource> Make(
    const std::string &url,
    bool legacyImplementation = false,
    bool acceptSelfSigned = false);

File: vnext/Shared/WinRTWebSocketResource.cpp [hunk 3]
```diff
@@ -86,7 +86,7 @@ namespace
 namespace Microsoft::React
 {
 
-WinRTWebSocketResource::WinRTWebSocketResource(IMessageWebSocket&& socket, IDataWriter&& writer, Uri&& uri, vector<ChainValidationResult> &&certExeptions)
+WinRTWebSocketResource::WinRTWebSocketResource(IMessageWebSocket&& socket, IDataWriter&& writer, Uri&& uri, vector<ChainValidationResult> &&certExeptions) noexcept
   : m_uri{ std::move(uri) }
   , m_socket{ std::move(socket) }
   , m_writer{ std::move(writer) }
```
Comment (NikoAri) [inline vnext/Shared/WinRTWebSocketResource.cpp:92]: (nit) usually, these follow same order as args for readability eg:
: m_socket{ std::move(socket) }
, m_writer{ std::move(writer) }
, m_uri{ std::move(uri) }

File: vnext/Shared/WinRTWebSocketResource.h (hunk 1 above)
Comment (JunielKatarn) [reply vnext/Shared/WinRTWebSocketResource.h:73]: That's clang format. We split at column 120, I believe.

File: vnext/Shared/IWebSocketResource.h (hunk 2 above)
Comment (JunielKatarn) [reply vnext/Shared/IWebSocketResource.h:87]: Decided not to. We currently pass error info through the existing exception if resource creation fails.

May revisit later.

File: vnext/Shared/WinRTWebSocketResource.cpp (hunk 3 above)
Comment (JunielKatarn) [reply vnext/Shared/WinRTWebSocketResource.cpp:92]: Originally, I think I chose order of declaration in the header, but can also change this in a subsequent PR.

File: vnext/Shared/IWebSocketResource.h (hunk 2 above)
Comment (JunielKatarn) [reply vnext/Shared/IWebSocketResource.h:87]: Not sure if clang format will take it, but will also remove one, or maybe both boolean args now that "feature gates" are supported.

**Pull request merged**